tools/run_tests/jobset.py

```diff
@@ -194,6 +194,7 @@ def state(self, update_cache):
       self._tempfile.seek(0)
       stdout = self._tempfile.read()
       filtered_stdout = filter(lambda x: x in string.printable, stdout.decode(errors='ignore'))
+      filtered_stdout = filtered_stdout[-128:0]
```

Comment:
should it be just [-128:]

please add a comment "# TODO: looks like jenkins master is slow because parsing the junit results XMLs is not implemented efficiently. This is an experiment to workaround the issue by making sure results.xml file is small enough."